# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/WindowContext.cpp
# Commit: 44e715808901
# Full Hash: 44e71580890195d163912b96f41ef09e32570ec8
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2020-06-13 03:21:52
# Regressor Bug: 1587743
# File Overlap Count: 1
# Description:
#   Bug 1587743 - Part 1: Pre-compute the delegated permissions for the top-level content and store it in the WindowContext. r=baku,nika
#   
#   In order to delegate the permission to the top-level window, in this
#   patch, we pre-compute the permissions of the top-level context and set
#   them to the top-level WindowContext. So, the cross-origin iframe can
# ==============================================================================

diff -r a890be633ec3 -r 44e715808901 docshell/base/WindowContext.cpp
--- a/docshell/base/WindowContext.cpp	Fri Jun 12 16:18:15 2020 +0000
+++ b/docshell/base/WindowContext.cpp	Fri Jun 12 16:31:49 2020 +0000
@@ -166,6 +166,20 @@
   return CheckOnlyOwningProcessCanSet(aSource);
 }
 
+bool WindowContext::CanSet(
+    FieldIndex<IDX_DelegatedPermissions>,
+    const PermissionDelegateHandler::DelegatedPermissionList& aValue,
+    ContentParent* aSource) {
+  return CheckOnlyOwningProcessCanSet(aSource);
+}
+
+bool WindowContext::CanSet(
+    FieldIndex<IDX_DelegatedExactHostMatchPermissions>,
+    const PermissionDelegateHandler::DelegatedPermissionList& aValue,
+    ContentParent* aSource) {
+  return CheckOnlyOwningProcessCanSet(aSource);
+}
+
 void WindowContext::CreateFromIPC(IPCInitializer&& aInit) {
   MOZ_RELEASE_ASSERT(XRE_IsContentProcess(),
                      "Should be a WindowGlobalParent in the parent");